design/ceph/object-store-realm.md

```diff
@@ -0,0 +1,169 @@
+
+## Prerequisites
+
+A rook ceph cluster without object-stores.
+
+## Ceph Multisite Overview
+
+Multisite is a feature of Ceph that allows object stores to replicate its data over multiple Ceph clusters. 
+
+Multisite also allows object stores to be independent and isloated from other object stores in a cluster.
+
+### Ceph Multisite data model
+
+For reference, here is a description of the underlying Ceph Multisite data model.
+
+```
+A cluster has one or more realms
+
+A realm spans one or more clusters
+A realm has one or more zonegroups
+A realm has one master zonegroup
+A realm defined in another cluster is replicated with the pull command
+The objects in a realm are independent and isolated from objects in other realms
+
+A zonegroup has one or more zones
+A zonegroup has one master zone
+A zonegroup spans one or more clusters
+A zonegroup defined in another cluster is replicated with the pull command
+A zonegroup defines a namespace for object IDs unique across its zones
+Zonegroup metadata is replicated to other zonegroups in the realm
+
+A zone belongs to one cluster
+A zone has a set of pools that store the user and object metadata and object data
+Zone data and metadata is replicated to other zones in the zonegroup
+A master zone needs to be created for secondary zones to pull from to replicate across zones
+```
+### Overview Ceph Multisite Steps
+To enable Ceph's multisite, the following steps need to happen.
+
+1. A realm needs to be created
+2. A master zonegroup in the realm needs to be created
+3. A master zone in the master zonegroup needs to be created
+4. An object store needs to be added to the master zone
+
+## Object Store Realm Walkthrough
+
+The resource descibed in this design document represents the realm in the Ceph Multisite data model.
+
+### Creating an Object Store Realm
+
+#### Config
+
+When the storage admin is ready to create a multisite realm for object storage, the admin will name the realm in the metadata section on the configuration file.
+
+This example `object-store-realm.yaml`, names a realm `my-realm`.
+```yaml
+apiVersion: ceph.rook.io/v1alpha1
+kind: CephObjectStoreRealm
+metadata:
+  name: my-realm
+  namespace: rook-ceph
+spec:
+```
+
+Now create the object store realm.
+```bash
+kubectl create -f object-store-realm.yaml
+```
+
+#### Steps
+
+1. At this point the Rook operator recognizes that a new object store realm resource needs to be configured. The operator will start creating the resource to start the object store realm.
+
+2. After these steps the admin should start up a master [object-store-zonegroup](/design/ceph/object-store-zonegroup.md), a master [object-store-zone](/design/ceph/object-store-zone.md), and a [object-store](/design/ceph/object-store.md) in the master object-store-zone resource.
+
+3. Once all of the resources from step #2 are created, the operator will create a realm on the Rook Ceph cluster and the object-realm-resource resource will be running.
```

Comment:
@travisn Do you think this is worded clearly enough to indicate that the resource won't be running until endpoints are in the master zone via creating the resources in step 2?